WooCommerce/Classes/ViewRelated/Dashboard/Settings/Privacy/PrivacyBannerPresentationUseCase.swift

```diff
@@ -1,30 +1,82 @@
 import Foundation
 import Yosemite
+import CoreTelephony
 
 /// Type who informs us if the privacy banner should be shown or not.
 ///
-struct PrivacyBannerPresentationUseCase {
-
-    /// Users current location country code.
-    ///
-    private let countryCode: String
+final class PrivacyBannerPresentationUseCase {
 
     /// User Defaults database
     ///
     private let defaults: UserDefaults
 
-    init(countryCode: String, defaults: UserDefaults = UserDefaults.standard) {
-        self.countryCode = countryCode
+    /// Stores to fetch remote information.
+    ///
+    private let stores: StoresManager
+
+    /// Users current locale.
+    ///
+    private let currentLocale: Locale
+
+    init(defaults: UserDefaults = UserDefaults.standard, stores: StoresManager = ServiceLocator.stores, currentLocale: Locale = .autoupdatingCurrent) {
         self.defaults = defaults
+        self.stores = stores
+        self.currentLocale = currentLocale
     }
 
     /// Returns `true` if the privacy banner should be shown.
     /// Currently it is shown if the user is in the EU zone & privacy choices have not been saved.
     ///
-    func shouldShowPrivacyBanner() -> Bool {
-        let isCountryInEU = Country.GDPRCountryCodes.contains(countryCode)
-        let hasSavedPrivacySettings = defaults.hasSavedPrivacyBannerSettings
-        return isCountryInEU && !hasSavedPrivacySettings
+    func shouldShowPrivacyBanner() async -> Bool {
+        do {
+            let countryCode = try await fetchUsersCountryCode()
+            let isCountryInEU = Country.GDPRCountryCodes.contains(countryCode)
+            let hasSavedPrivacySettings = defaults.hasSavedPrivacyBannerSettings
+            return isCountryInEU && !hasSavedPrivacySettings
+        } catch {
+            DDLogInfo("⛔️ Could not determine users country code. Error: \(error)")
+            return false
+        }
+    }
+}
+
+// MARK: Private Helpers
+private extension PrivacyBannerPresentationUseCase {
+
+    /// Determines the user country code by the following algorithm.
+    /// - If the user has a WPCOM account:
+    ///   - Use the ip country code.
+    /// - If the user does not has a WPCOM account:
+    ///   - Use the current locale country code.
+    ///
+    func fetchUsersCountryCode() async throws -> String {
+        // Use ip country code for WPCom accounts
+        if !stores.isAuthenticatedWithoutWPCom {
+            return try await fetchIPCountryCode()
+        }
+
+        // Use locale country code as a fallback
+        return fetchLocaleCountryCode()
+    }
```

Comment:
Do you think it's worth using `fetchLocaleCountryCode()` as a fallback if `fetchIPCountryCode()` throws an error (not just when the user is not authenticated with WPCOM)?